File: algorithms/python/createTargetArrayintheGivenOrder/createTargetArrayintheGivenOrder.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/create-target-array-in-the-given-order/
# Author: Miao Zhang
# Date:   2021-04-28

class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        res = []
        for i in range(len(nums)):
            res.insert(index[i], nums[i])
        return res

File: algorithms/cpp/createTargetArrayintheGivenOrder/createTargetArrayintheGivenOrder.cpp
// Source: https://leetcode.com/problems/create-target-array-in-the-given-order/
// Author: Miao Zhang
// Date:   2021-04-28

class Solution {
public:
    vector<int> createTargetArray(vector<int>& nums, vector<int>& index) {
        vector<int> res;
        for (int i = 0; i < nums.size(); i++) {
            res.insert(begin(res) + index[i], nums[i]);
        }
        return res;
    }
};
